### DISCUSSION TITLE
Reset 1614 ??

### DISCUSSION THREAD

**Discussion by stevecolt** [SpenceKonde/megaTinyCore#478] [Q&A]
Hello
theres a lot more to these little chips than I realised!
I am using a curiosity nano as UDPI programmer, and was using as serial comms for debug, though as this doesn't reset after an upload means that comms was very flaky. The only way to get serial working is to use an FTDI adapter and upload with it connected. This really isn't a great solution.
So was looking at implementing reset, simplest looking like a switch/DTR on an input, and if so why not use the UDPI pin as the Reset pin? 
I have tried it using another pin as a test, which unfortunately didn't work
void resetViaSWR() {
  _PROTECTED_WRITE(RSTCTRL.SWRR, 1);
}
According to visual studio  _PROTECTED_WRITE is undefined but doesn't throw any errors. The data sheet does say if it isn't in protected mode it wont work! I'm guessing I need another include, but which one?
Can the UDPI be used as an input pin after programming in UDPI mode, I cant find anything that says it cant?
Thanks
Steve

**Comment by SpenceKonde** [comment]
Your syntax is correct; SWRR definitely works, quite well in fact. So the problem is with parts of your code you didn't include in the post. 

I thought that PA0 worked as an input when still set as UPDI, though I vaguely recall having problems trying to do exactly what you're doing now. I generally leave a serial adapter connected whenever I'm quickly iterating code. 

See also the ErsatzReset example, which demonstrates using as an ersatz reset pin, including a LOW level-sensing interrupt. I think there's also a section of the readme that talks about resetting the sketch. 

Sounds like VisualStudio doesn't know about the io header files AND doesn't know enough to use the core's
